packages/core/src/common/utils.ts

```diff
@@ -185,10 +185,10 @@ export function clamp(val: number, min: number, max: number) {
 
 /** Returns the number of decimal places in the given number. */
 export function countDecimalPlaces(num: number) {
-    if (typeof num !== "number" || Math.floor(num) === num) {
-        return 0;
-    }
-    return num.toString().split(".")[1].length;
+    if (!isFinite(num)) return 0;
+    let e = 1, p = 0;
+    while (Math.round(num * e) / e !== num) { e *= 10; p++; }
```

Comment:
this is very neat but I have some performance concerns. need to inspect how often this function is executed.

Reply:
ah it's only used to update state so should be alright 👍